src/lex/mod.rs

```diff
@@ -374,31 +373,21 @@ impl Lexer {
             self.next_char();
         }
         let float = if hex {
-            hexf_parse::parse_hexf64(&buf, false).map_err(|err| err.to_string())
-        } else {
-            buf.parse()
-                .map_err(|err: std::num::ParseFloatError| err.to_string())
-        }?;
-        if float.is_infinite() {
-            return Err("overflow parsing floating literal".into());
-        }
-        let should_be_zero = buf.bytes().all(|c| match c {
-            b'.' | b'+' | b'-' | b'e' | b'p' | b'0' => true,
-            _ => false,
-        });
-        if float == 0.0 && !should_be_zero {
-            Err("underflow parsing floating literal".into())
+            // Leaving as generic because it will be replaced shortly.
+            hexf_parse::parse_hexf64(&buf, false)
+                .map_err(|err| LexError::Generic(err.to_string()))?
         } else {
-            Ok(float)
-        }
+            buf.parse::<f64>()?
+        };
+        Ok(float)
     }
     // returns None if there are no digits at the current position
     fn parse_int(
         &mut self,
         mut acc: u64,
         radix: u32,
```

Comment:
Same as above, it would be nice to take a `Radix` instead of a u32 to avoid the unwrap.